design/src/rfcs/rfc0012_bytestream_callback_api.md

```diff
@@ -0,0 +1,112 @@
+RFC: A callback API for `ByteStream`
+====================================
+
+> Status: RFC
+
+Adding a callback API to `ByteStream` will enable developers using the SDK to implement things like checksum validations and 'read progress' callbacks.
```

Comment:
The "read progress" use-case is well covered by the doc, but there's little detail about enabling the checksum use-case.

Reply:
I added a possible request checksum implementation example in the latest update.